packages/client-core/src/admin/common/Location/LocationDrawer.tsx

```diff
@@ -181,8 +194,54 @@ const LocationDrawer = ({ open, mode, selectedLocation, selectedScene, onClose }
     state.merge({ [name]: value })
   }
 
-  const handleSubmit = () => {
-    const data: LocationData = {
+  const handleSubmit = async () => {
+    confirmWindowOpen.set(false)
+
+    state.formErrors.merge({
+      name: state.name.value ? '' : t('admin:components.location.nameCantEmpty'),
+      maxUsers: state.maxUsers.value ? '' : t('admin:components.location.maxUserCantEmpty'),
+      scene: state.scene.value ? '' : t('admin:components.location.sceneCantEmpty'),
+      type: state.type.value ? '' : t('admin:components.location.typeCantEmpty')
+    })
+
+    if (!validateForm(state.value, state.formErrors.value)) {
+      NotificationService.dispatchNotify(t('admin:components.common.fillRequiredFields'), { variant: 'error' })
+      return
+    }
+
+    // TODO: new checkbox
+    const isBaked = true
+
+    // TODO: Present blocking modal to user "Are you sure you want to Publish?"
+
+    await publishScene(isBaked)
+
+    // TODO: After publish succeeds or fails, restore original scene and close blocking modal
+
+    handleClose()
+  }
+
+  const publishScene = async (bake: boolean) => {
+    // TODO: Save current scene in place, and create new version to publish
+    const { sceneAssetID, projectName, sceneName, scenePath } = getState(EditorState)
+
+    if (projectName && sceneName && scenePath) {
+      const sceneQuery = useFind(assetPath, { query: { assetURL: scenePath ?? '' } }).data
+      const sceneURL = sceneQuery?.[0]?.assetURL
+      const abortController = new AbortController()
+
+      // TODO: Bake implementation
+      if (bake) {
+        await saveSceneGLTF(sceneAssetID, projectName, sceneName + '-optimized', abortController.signal)
+      }
+
+      setCurrentEditorScene(sceneURL, sceneQuery[0].id! as EntityUUID)
```

Comment:
Also, this needs to be in the "bake" conditional as well, because if we're not baking we have no need to change editor scene